appserver/website-bootstrap/explore.js

```diff
@@ -1,97 +1,250 @@
+                    cell1.innerHTML = trends[i].trend_content;
+                    let cell2 = row.insertCell(1);
+                    cell2.innerHTML = trends[i].tweet_volume.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","); // adds commas every third digit for numbers greater than 999
+                }
+            });
+        } else {
+            alert("Please provide a valid city name!");
+        }
+
+    });
+
+    document.querySelector("#trends-display-tweets").addEventListener('click', function() {
+        let tweetTrendUrl = 'http://18.214.197.203:5000/toptweets?query=';
+        let trend = document.getElementById("tweet-trend").value;
+        let tweetTrend = encodeURIComponent(trend); // so we can search for trends with hashtags
+        let tweets = null;
+        if(trend){
+            tweetTrendUrl += tweetTrend;
+            $.getJSON(tweetTrendUrl, function(data){
+		        tweets = data;
+	        }).then(function() {
+	            document.getElementById("tweets-title").innerHTML = "Recent Tweets for " + trend;
+                let table = document.getElementById("display-tweets-table");
+                for(let i = 0; i < tweets.length; i++){
+                    let row = table.insertRow(i+1);
+                    let cell1 = row.insertCell(0);
+                    cell1.innerHTML = tweets[i].content;
+                    let cell2 = row.insertCell(1);
+                    cell2.innerHTML = tweets[i].username;
+                    let cell3 = row.insertCell(2);
+                    let t_date = (tweets[i].tweet_date).substring(0, 20);
+                    cell3.innerHTML = t_date;
+                    let cell4 = row.insertCell(3);
+                    cell4.innerHTML = tweets[i].retweets.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","); // adds commas every third digit for numbers greater than 999
+                    let cell5 = row.insertCell(4);
+                    cell5.innerHTML = tweets[i].likes.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","); // adds commas every third digit for numbers greater than 999
+                }
+            });
+
+        } else {
+            alert("Please provide a trend!");
+        }
+    });
+
+    document.querySelector("#trends-display-city-time").addEventListener('click', function() {
+        let cityTrendUrl = 'http://18.214.197.203:5000/temporal?woeid=';
+        let city = document.getElementById('myInput2').value;
+        let woeid = citiesMap.get(city);
+        cityTrendUrl += woeid;
+        let trends = null;
+        if(woeid){
+            cityTrendUrl += woeid;
+            let fromDate = document.getElementById('start').value;
+            let toDate = document.getElementById('end').value;
+            if(!fromDate || !toDate){
+                alert("Please provide start and ends dates!");
+            } else {
+                let startDate = new Date(fromDate);
+                let startBoundary = new Date("02/29/2020");
```

Comment:
Does this work? I'm concerned, based on looking at the Date() documentation, that this might not be accepted.

Reply:
for the startBoundary?

Reply:
maybe creating a new Date object with individual numbers for individual parameters, like option 4 in this documentation?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

Reply:
and also for the fromDate user input, if they'd be entering strings in the MM/DD/TYYY format. If they are, I suggest tokenizing the input string based on the '/' character and then verifying that that produces 3 tokens which can be parsed as strings.

Reply:
you want something like this: 
new Date(year, monthIndex [, day [, hours [, minutes [, seconds [, milliseconds]]]]])

instead of let startBoundary = new Date("02/29/2020");?

Reply:
yes, using the year, monthIndex, and day parameters. Unless I'm misunderstanding the Date documentation and that string is parsed correctly when fed into the constructor?